File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    # 서로 다른 종류의 수를 최소화
    answer = cnt = 0
    tDict = {}
    for t in tangerine:
        if t in tDict:
            tDict[t] += 1
        else:
            tDict[t] = 1
    sorted_tDict = sorted(tDict.items(), key=lambda x: -x[1])
    idx = 0
    while True:
        if cnt >= k:
            break
        cnt += sorted_tDict[idx][1]
        idx += 1
        answer += 1
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    let cnt = 0;
    let tDict = {};
    for (let i = 0; i < tangerine.length; i++) {
        if (tangerine[i] in tDict) {
            tDict[tangerine[i]] += 1
        } else {
            tDict[tangerine[i]] = 1
        }
    }
    let tmp = Object.entries(tDict)
    tmp.sort((a,b) => b[1] - a[1])
    for (t of tmp) {
        if (cnt >= k) {
            break
        }
        cnt += t[1]
        answer += 1
    }
    return answer;
}